go/mysql/replication/replication_status.go

```diff
@@ -177,30 +177,33 @@ func ProtoToReplicationStatus(s *replicationdatapb.Status) ReplicationStatus {
 	return replstatus
 }
 
-// FindErrantGTIDs can be used to find errant GTIDs in the receiver's relay log, by comparing it against all known replicas,
-// provided as a list of ReplicationStatus's. This method only works if the flavor for all retrieved ReplicationStatus's is MySQL.
-// The result is returned as a Mysql56GTIDSet, each of whose elements is a found errant GTID.
-// This function is best effort in nature. If it marks something as errant, then it is for sure errant. But there may be cases of errant GTIDs, which aren't caught by this function.
-func (s *ReplicationStatus) FindErrantGTIDs(otherReplicaStatuses []*ReplicationStatus) (Mysql56GTIDSet, error) {
-	if len(otherReplicaStatuses) == 0 {
+// LegacyFindErrantGTIDs is the legacy implementation for FindErrantGTIDs. It has an additional check for
+// checking if one of the replicas is the superset of the other if only 2 are available. We need this check
+// only for the situation where we don't have the `ReadReparentJournalInfo` RPC implemented in the vttablet.
+func LegacyFindErrantGTIDs(position Position, sourceUUID SID, otherPositions []Position) (Mysql56GTIDSet, error) {
```

Comment:
What happens if we delete the legacy handling on main, i.e. v22? Given that we have added the RPC to v21, the new code will take effect as soon as vtctld or vtorc is upgraded to v22, and it will work even if vttablets are still on v21.

Reply:
Yes, I removed the legacy handling.